controllers/UserController.js

```diff
@@ -0,0 +1,100 @@
+import UserModel from '../models/User.js'
+import jwt from 'jsonwebtoken'
+import bcrypt from 'bcrypt'
+import { validationResult } from 'express-validator';
+
+export const register = async (req, res) => {
+    try {
+        const errors = validationResult(req);
+        if(!errors.isEmpty()) {
+            return res.status(400).json(errors.array());
+        }
+
+        const password = req.body.password;
+        const salt = await bcrypt.genSalt(10);
```

Comment:
You can introduce a constant saltRounds to specify the number of salt rounds for bcrypt's genSalt method:
```suggestion
        const saltRounds = 10;
        const salt = await bcrypt.genSalt(saltRounds);
```